# Worksheet 7

In [4]:
import numpy as np

# I'll create an array to use below
A = np.random.randint(100, size=(10, 4))

 # 1. Write a function that takes an array and returns the column sums of this array.
 
 Hint: use "axis = " argument would make things much easier

In [5]:
def getSum(array):
    return np.sum(array, axis=0)

In [6]:
# you could print out the result of getSum(A)
print(getSum(A))

[455 597 455 353]


You can check your result by printing out A : print(A)

# 3. Now use your function to get the column sums of this array.

You should not have to modify your function.

In [7]:
B = np.array(      [ [13, 92, 23, 25, 77, 11],
                     [28, 15, 18, 56, 45, 51],
                     [10, 83, 33, 30, 49, 74],
                     [33, 49, 76, 44, 48, 75],
                     [24, 18, 29, 28, 82, 77],
                     [14, 22, 83, 52, 66, 85],
                     [89, 37, 71, 73, 83, 28]]     )

What are the column sums of this array?

In [8]:
getSum(B)

array([211, 316, 333, 308, 450, 401])

The result I get is [211 316 333 308 450 401].

# 4. Now write a function that determines what fraction of the given data set is above its mean. 

Again, your function should work for any array, not just the one given below.

In [17]:
def aboveMean(data):
    return np.sum(data>np.mean(data))/len(data)

In [18]:
C = np.array([42,100,131,8,3,2,12,10,9,5,12,20,13,4,1,11,9,5,88,30,124,66,34,78,20])

print(aboveMean(C))

0.32


I have 0.32

In [19]:
import comp116
comp116.submit('ws7')